src/query/execute_statement.c

```diff
@@ -17246,6 +17262,651 @@ do_set_timezone (PARSER_CONTEXT * parser, PT_NODE * statement)
+{
+  MOP synonym_object = NULL;
+
+  int error = NO_ERROR;
+  int save = 0;
+
+  synonym_object = do_get_synonym (name, owner_name, is_public);
+  if (synonym_object == NULL)
+    {
+      if (if_exists)
+	{
+	  return NO_ERROR;
+	}
+
+      assert (er_errid () != NO_ERROR);
+      error = er_errid ();
+      goto end;
+    }
+
+  error = au_check_synonym_authorization (synonym_object);
+  if (error < NO_ERROR)
+    {
+      if (error == ER_QPROC_CANNOT_UPDATE_SYNONYM)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 2, owner_name, name);
+	}
+
+      goto end;
+    }
+
+  AU_DISABLE (save);
+
+  error = db_drop (synonym_object);
+
+  AU_ENABLE (save);
+
+end:
+  return error;
+}
+
+/*
+ * do_rename_synonym () - rename synonym.
+ * return: error code
+ * parser(in): parser context
+ * statement(in): parse tree of a statement
+ */
+int
+do_rename_synonym (PARSER_CONTEXT * parser, PT_NODE * statement)
+{
+  int error = ER_QPROC_NOT_SUPPORT;
+  er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, error, 1, "RENAME SYNONYM");
+
+  /* To do. */
+
+  return error;
+}
+
+/*
+ * do_get_synonym () - get a synonym object pointer.
+ * return: synonym object pointer
+ * name(in): synonym namess
+ * owner_name(in): synonym owner name
+ * is_public(in): access modifier
+ */
+MOP
+do_get_synonym (const char *name, const char *owner_name, const int is_public)
+{
+#define QUERY_BUF_SIZE 2048
+
+  const char *query = NULL;
+  char query_buf[QUERY_BUF_SIZE];
+  DB_QUERY_RESULT *query_result = NULL;
+  DB_QUERY_ERROR query_error;
+  MOP synonym_object = NULL;
+
+  int error = NO_ERROR;
+  int save = 0;
+
+  memset (&query_buf, 0, sizeof (char) * QUERY_BUF_SIZE);
+  memset (&query_error, 0, sizeof (DB_QUERY_ERROR));
+
+  query = "SELECT [%s] FROM [%s] WHERE [name] = '%s' AND [owner].[name] = '%s' AND [is_public] = %d LIMIT 1";
+  error = sprintf (query_buf, query, CT_SYNONYM_NAME, CT_SYNONYM_NAME, name, owner_name, is_public);
+  if (error < NO_ERROR)
+    {
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, error, 0);
+      goto end;
+    }
+
+  AU_DISABLE (save);
+
+  error = db_compile_and_execute_local (query_buf, &query_result, &query_error);
+
+  AU_DISABLE (save);
+
+  if (error < NO_ERROR)
+    {
+      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, error, 0);
+      goto end;
+    }
+
+  if (db_query_first_tuple (query_result) == DB_CURSOR_SUCCESS)
+    {
+      DB_VALUE value;
+      db_make_null (&value);
+
+      if (db_query_get_tuple_value (query_result, 0, &value) == NO_ERROR)
+	{
+	  synonym_object = db_get_object (&value);
+	}
+    }
+  else
+    {
+      if (is_public)
+	{
+	  er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_QPROC_PUBLIC_SYNONYM_NOT_FOUND, 2, owner_name, name);
+	  synonym_object = NULL;
+	  goto end;
+	}
+      else
+	{
+	  er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_QPROC_PRIVATE_SYNONYM_NOT_FOUND, 2, owner_name, name);
+	  synonym_object = NULL;
+	  goto end;
+	}
```

Comment:
suggested shorten version. The decision is yours.
```suggestion
      error = is_public ? ER_QPROC_PUBLIC_SYNONYM_NOT_FOUND : ER_QPROC_PRIVATE_SYNONYM_NOT_FOUND;
      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, error, 2, owner_name, name);
	synonym_object = NULL;
	goto end;
```

Reply:
Thank you.
Thank you.
Thank you.

I didn't think of the triple.
I will change it.